Tools/tsan/supressions.txt

```diff
@@ -3,3 +3,5 @@ race:get_allocator_unlocked
 race:set_allocator_unlocked
 race:mi_heap_visit_pages
 race:_mi_heap_delayed_free_partial
+
+thread:pthread_create
```

Comment:
Due to thread leak at the default build

```
0:00:41 load avg: 2.89 [12/14/1] test_logging failed (1 failure) -- running (1): test_io (30.7 sec)
/home/runner/work/cpython/cpython/Lib/test/test_logging.py:783: DeprecationWarning: This process (pid=11100) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()
==================
WARNING: ThreadSanitizer: thread leak (pid=13471)
  Thread T36 (tid=13461, finished) created by main thread at:
    #0 pthread_create <null> (python+0xc9fed) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #1 do_start_joinable_thread /home/runner/work/cpython/cpython/Python/thread_pthread.h:284:14 (python+0x5455b8) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #2 PyThread_start_joinable_thread /home/runner/work/cpython/cpython/Python/thread_pthread.h:308:9 (python+0x5453da) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #3 do_start_new_thread /home/runner/work/cpython/cpython/./Modules/_threadmodule.c:1430:15 (python+0x5f2f8b) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #4 thread_PyThread_start_joinable_thread /home/runner/work/cpython/cpython/./Modules/_threadmodule.c:1521:9 (python+0x5f209b) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #5 _PyEval_EvalFrameDefault /home/runner/work/cpython/cpython/Python/generated_cases.c.h:1181:23 (python+0x41f2ba) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #6 _PyEval_EvalFrame /home/runner/work/cpython/cpython/./Include/internal/pycore_ceval.h:114:16 (python+0x417127) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #7 _PyEval_Vector /home/runner/work/cpython/cpython/Python/ceval.c:1820:12 (python+0x417127)
    #8 _PyFunction_Vectorcall /home/runner/work/cpython/cpython/Objects/call.c (python+0x23b6c7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #9 _PyObject_VectorcallTstate /home/runner/work/cpython/cpython/./Include/internal/pycore_call.h:168:11 (python+0x240945) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #10 method_vectorcall /home/runner/work/cpython/cpython/Objects/classobject.c:92:18 (python+0x23f349) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #11 _PyVectorcall_Call /home/runner/work/cpython/cpython/Objects/call.c:273:16 (python+0x23b0af) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #12 _PyObject_Call /home/runner/work/cpython/cpython/Objects/call.c:348:16 (python+0x23b312) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #13 PyObject_Call /home/runner/work/cpython/cpython/Objects/call.c:373:12 (python+0x23b4f7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #14 _PyEval_EvalFrameDefault /home/runner/work/cpython/cpython/Python/generated_cases.c.h:1270:26 (python+0x41fa21) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #15 _PyEval_EvalFrame /home/runner/work/cpython/cpython/./Include/internal/pycore_ceval.h:114:16 (python+0x417127) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #16 _PyEval_Vector /home/runner/work/cpython/cpython/Python/ceval.c:1820:12 (python+0x417127)
    #17 _PyFunction_Vectorcall /home/runner/work/cpython/cpython/Objects/call.c (python+0x23b6c7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #18 _PyObject_VectorcallDictTstate /home/runner/work/cpython/cpython/Objects/call.c:135:15 (python+0x239dac) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #19 _PyObject_Call_Prepend /home/runner/work/cpython/cpython/Objects/call.c:504:24 (python+0x23badf) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #20 slot_tp_call /home/runner/work/cpython/cpython/Objects/typeobject.c:9176:15 (python+0x336d05) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #21 _PyObject_MakeTpCall /home/runner/work/cpython/cpython/Objects/call.c:242:18 (python+0x23a1b5) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[22](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:23) _PyObject_VectorcallTstate /home/runner/work/cpython/cpython/./Include/internal/pycore_call.h:166:16 (python+0x[23](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:24)9b06) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #23 PyObject_Vectorcall /home/runner/work/cpython/cpython/Objects/call.c:327:12 (python+0x23b1c0) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[24](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:25) _PyEval_EvalFrameDefault /home/runner/work/cpython/cpython/Python/generated_cases.c.h:817:23 (python+0x41cf94) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[25](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:26) _PyEval_EvalFrame /home/runner/work/cpython/cpython/./Include/internal/pycore_ceval.h:114:16 (python+0x417127) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[26](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:27) _PyEval_Vector /home/runner/work/cpython/cpython/Python/ceval.c:1820:12 (python+0x4171[27](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:28))
    #27 _PyFunction_Vectorcall /home/runner/work/cpython/cpython/Objects/call.c (python+0x23b6c7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[28](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:29) _PyObject_VectorcallTstate /home/runner/work/cpython/cpython/./Include/internal/pycore_call.h:168:11 (python+0x240945) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[29](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:30) method_vectorcall /home/runner/work/cpython/cpython/Objects/classobject.c:92:18 (python+0x23f349) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[30](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:31) _PyVectorcall_Call /home/runner/work/cpython/cpython/Objects/call.c:273:16 (python+0x23b0af) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[31](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:32) _PyObject_Call /home/runner/work/cpython/cpython/Objects/call.c:348:16 (python+0x23b312) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[32](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:33) PyObject_Call /home/runner/work/cpython/cpython/Objects/call.c:373:12 (python+0x23b4f7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[33](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:34) _PyEval_EvalFrameDefault /home/runner/work/cpython/cpython/Python/generated_cases.c.h:1270:26 (python+0x41fa21) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[34](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:35) _PyEval_EvalFrame /home/runner/work/cpython/cpython/./Include/internal/pycore_ceval.h:114:16 (python+0x417127) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[35](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:36) _PyEval_Vector /home/runner/work/cpython/cpython/Python/ceval.c:1820:12 (python+0x417127)
    #[36](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:37) _PyFunction_Vectorcall /home/runner/work/cpython/cpython/Objects/call.c (python+0x23b6c7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[37](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:38) _PyObject_VectorcallDictTstate /home/runner/work/cpython/cpython/Objects/call.c:135:15 (python+0x239dac) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[38](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:39) _PyObject_Call_Prepend /home/runner/work/cpython/cpython/Objects/call.c:504:24 (python+0x23badf) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[39](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:40) slot_tp_call /home/runner/work/cpython/cpython/Objects/typeobject.c:9176:15 (python+0x336d05) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[40](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:41) _PyObject_MakeTpCall /home/runner/work/cpython/cpython/Objects/call.c:242:18 (python+0x23a1b5) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[41](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:42) _PyObject_VectorcallTstate /home/runner/work/cpython/cpython/./Include/internal/pycore_call.h:166:16 (python+0x239b06) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[42](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:43) PyObject_Vectorcall /home/runner/work/cpython/cpython/Objects/call.c:327:12 (python+0x23b1c0) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[43](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:44) _PyEval_EvalFrameDefault /home/runner/work/cpython/cpython/Python/generated_cases.c.h:817:23 (python+0x41cf94) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[44](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:45) _PyEval_EvalFrame /home/runner/work/cpython/cpython/./Include/internal/pycore_ceval.h:114:16 (python+0x417127) (BuildId: e2d2ad1613e99b61d[45](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:46)0ee5f7[46](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:47)96ad3a0e5d6f7)
    #45 _PyEval_Vector /home/runner/work/cpython/cpython/Python/ceval.c:1820:12 (python+0x417127)
    #46 _PyFunction_Vectorcall /home/runner/work/cpython/cpython/Objects/call.c (python+0x23b6c7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[47](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:48) _PyObject_VectorcallTstate /home/runner/work/cpython/cpython/./Include/internal/pycore_call.h:168:11 (python+0x240945) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[48](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:49) method_vectorcall /home/runner/work/cpython/cpython/Objects/classobject.c:92:18 (python+0x23f3[49](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:50)) (BuildId: e2d2ad1613e99b61d4[50](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:51)ee5f74696ad3a0e5d6f7)
    #49 _PyVectorcall_Call /home/runner/work/cpython/cpython/Objects/call.c:273:16 (python+0x23b0af) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #50 _PyObject_Call /home/runner/work/cpython/cpython/Objects/call.c:348:16 (python+0x23b312) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[51](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:52) PyObject_Call /home/runner/work/cpython/cpython/Objects/call.c:373:12 (python+0x23b4f7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[52](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:53) _PyEval_EvalFrameDefault /home/runner/work/cpython/cpython/Python/generated_cases.c.h:1270:26 (python+0x41fa21) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[53](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:54) _PyEval_EvalFrame /home/runner/work/cpython/cpython/./Include/internal/pycore_ceval.h:114:16 (python+0x417127) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[54](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:55) _PyEval_Vector /home/runner/work/cpython/cpython/Python/ceval.c:1820:12 (python+0x417127)
    #[55](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:56) _PyFunction_Vectorcall /home/runner/work/cpython/cpython/Objects/call.c (python+0x23b6c7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[56](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:57) _PyObject_VectorcallDictTstate /home/runner/work/cpython/cpython/Objects/call.c:135:15 (python+0x239dac) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[57](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:58) _PyObject_Call_Prepend /home/runner/work/cpython/cpython/Objects/call.c:504:24 (python+0x23badf) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[58](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:59) slot_tp_call /home/runner/work/cpython/cpython/Objects/typeobject.c:9176:15 (python+0x336d05) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[59](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:60) _PyObject_MakeTpCall /home/runner/work/cpython/cpython/Objects/call.c:242:18 (python+0x23a1b5) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[60](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:61) _PyObject_VectorcallTstate /home/runner/work/cpython/cpython/./Include/internal/pycore_call.h:166:16 (python+0x239b06) (BuildId: e2d2ad1[61](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:62)3e99b61d450ee5f74696ad3a0e5d6f7)
    #61 PyObject_Vectorcall /home/runner/work/cpython/cpython/Objects/call.c:327:12 (python+0x23b1c0) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[62](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:63) _PyEval_EvalFrameDefault /home/runner/work/cpython/cpython/Python/generated_cases.c.h:817:23 (python+0x41cf94) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[63](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:64) _PyEval_EvalFrame /home/runner/work/cpython/cpython/./Include/internal/pycore_ceval.h:114:16 (python+0x416ee3) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[64](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:65) _PyEval_Vector /home/runner/work/cpython/cpython/Python/ceval.c:1820:12 (python+0x416ee3)
    #[65](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:66) PyEval_EvalCode /home/runner/work/cpython/cpython/Python/ceval.c:599:21 (python+0x416ee3)
    #[66](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:67) builtin_exec_impl /home/runner/work/cpython/cpython/Python/bltinmodule.c:1132:17 (python+0x4113f7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[67](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:68) builtin_exec /home/runner/work/cpython/cpython/Python/clinic/bltinmodule.c.h:521:20 (python+0x4113f7)
    #[68](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:69) cfunction_vectorcall_FASTCALL_KEYWORDS /home/runner/work/cpython/cpython/Objects/methodobject.c:441:24 (python+0x2ce744) (BuildId: e2d2ad1613e99b61d450ee5f74[69](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:70)6ad3a0e5d6f7)
    #69 _PyObject_VectorcallTstate /home/runner/work/cpython/cpython/./Include/internal/pycore_call.h:168:11 (python+0x239aa2) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[70](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:71) PyObject_Vectorcall /home/runner/work/cpython/cpython/Objects/call.c:327:12 (python+0x23b1c0) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[71](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:72) _PyEval_EvalFrameDefault /home/runner/work/cpython/cpython/Python/generated_cases.c.h:817:23 (python+0x41cf94) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[72](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:73) _PyEval_EvalFrame /home/runner/work/cpython/cpython/./Include/internal/pycore_ceval.h:114:16 (python+0x417127) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[73](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:74) _PyEval_Vector /home/runner/work/cpython/cpython/Python/ceval.c:1820:12 (python+0x417127)
    #[74](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:75) _PyFunction_Vectorcall /home/runner/work/cpython/cpython/Objects/call.c (python+0x23b6c7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[75](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:76) _PyVectorcall_Call /home/runner/work/cpython/cpython/Objects/call.c:273:16 (python+0x23b0af) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[76](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:77) _PyObject_Call /home/runner/work/cpython/cpython/Objects/call.c:348:16 (python+0x23b312) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[77](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:78) PyObject_Call /home/runner/work/cpython/cpython/Objects/call.c:373:12 (python+0x23b4f7) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[78](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:79) pymain_run_module /home/runner/work/cpython/cpython/Modules/main.c:297:14 (python+0x562460) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[79](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:80) pymain_run_python /home/runner/work/cpython/cpython/Modules/main.c (python+0x56183d) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[80](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:81) Py_RunMain /home/runner/work/cpython/cpython/Modules/main.c:707:5 (python+0x56183d)
    #[81](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:82) pymain_main /home/runner/work/cpython/cpython/Modules/main.c:737:12 (python+0x5622ea) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[82](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:83) Py_BytesMain /home/runner/work/cpython/cpython/Modules/main.c:761:12 (python+0x56235b) (BuildId: e2d2ad1613e99b61d450ee5f74696ad3a0e5d6f7)
    #[83](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:84) main /home/runner/work/cpython/cpython/./Programs/python.c:15:12 (python+0x148d4e) (BuildId: e2d2ad1613e99b61d450ee5f746[96](https://github.com/python/cpython/actions/runs/8300842825/job/22719646968#step:8:97)ad3a0e5d6f7)
```suggestion
thread:pthread_create
```
```

Reply:
We need to skip any tests that use threads after `fork()` when running with TSAN -- TSAN doesn't support it.

Here are some relevant commits from the `nogil-3.12` fork:
* https://github.com/colesbury/nogil-3.12/commit/c9fc49666cc7a3aa306a554c2b16751859821cb4
* https://github.com/colesbury/nogil-3.12/commit/7507a77a986e7fa16aab7ca7c7548a6dcfc46c6a
* https://github.com/colesbury/nogil-3.12/commit/a62d37674cf283b26970c3330838ca54f9d10310